packages/vsce/src/SshConfigUtils.ts

```diff
@@ -34,145 +39,262 @@ export class SshConfigUtils {
-        const migratedConfigs = (await ZClientUtils.migrateSshConfig()).filter(
+        // Get configs from ~/.ssh/config
+        const migratedConfigs = await ZClientUtils.migrateSshConfig();
+
+        // Parse to remove migratable configs that already exist on the team config
+        const filteredMigratedConfigs = migratedConfigs.filter(
             (migratedConfig) => !sshProfiles.some((sshProfile) => sshProfile.profile?.host === migratedConfig.hostname),
         );
 
+        // Choose between adding a new ssh host, an existing team config profile, and migrating from config.
         const qpItems: vscode.QuickPickItem[] = [
             { label: "$(plus) Add New SSH Host..." },
-            ...sshProfiles.map((prof) => ({ label: prof.name!, description: prof.profile?.host })),
-            { label: "Migrate From SSH Config", kind: vscode.QuickPickItemKind.Separator },
-            ...migratedConfigs.map((config) => ({ label: config.name!, description: config.hostname })),
+            ...sshProfiles.map(({ name, profile }) => ({
+                label: name!,
+                description: profile?.host,
+            })),
+            {
+                label: "Migrate From SSH Config",
+                kind: vscode.QuickPickItemKind.Separator,
+            },
+
+            ...filteredMigratedConfigs.map(({ name, hostname }) => ({
+                label: name!,
+                description: hostname,
+            })),
         ];
-        const result = await Gui.showQuickPick(qpItems, { title: "Choose an SSH host" });
-
-        let selectedProfile: sshConfigExt | undefined;
-
-        if (result === qpItems[0] || migratedConfigs.some((config) => result?.label === config.name!)) {
-            // Create new config
-            if (result === qpItems[0]) {
-                let newConfig = await SshConfigUtils.createNewConfig();
-                if (newConfig) {
-                    await profCache.refresh();
-                    if (newConfig && !newConfig.name) {
-                        newConfig = await SshConfigUtils.getNewProfileName(newConfig);
-                        if (!newConfig?.name) {
-                            vscode.window.showWarningMessage("SSH setup cancelled.");
-                            return;
+
+        const result = await Gui.showQuickPick(qpItems, {
+            title: "Choose an SSH host",
+        });
+
+        // If nothing selected, return
+        if (!result) return;
+
+        if (result === qpItems[0] || filteredMigratedConfigs.some(({ name }) => result.label === name)) {
+            const isNewConfig = result === qpItems[0];
+
+            // If result is add new SSH host then create a new config, if not use migrated configs
+            let selectedProfile = isNewConfig
+                ? await SshConfigUtils.createNewConfig()
+                : filteredMigratedConfigs.find(({ name }) => result.label === name);
+
+            if (!selectedProfile) return;
```

Comment:
No need to accept this suggestion. This is mostly for aesthetics 😋 

---

In theory we could avoid some indentation 😋 

```suggestion
        let selectedProfile = filteredMigratedConfigs.find(({ name }) => result.label === name);
        if (result === qpItems[0]) {
            // If result is add new SSH host then create a new config, if not use migrated configs
            selectedProfile = await SshConfigUtils.createNewConfig();
        }
        if (!selectedProfile) return;
```
Then every line that was inside the if condition could be moved back a few spaces 😋